### PR TITLE
8.0 add website doc jsonrpc

### PR DESCRIPTION
**Pull request by OSguard**
add module to display models and fields inside website

### CODE REVIEW

**Comment by OSguard** [comment]
On 10.08.2014 19:49, Nhomar Hernandez wrote (on mailing list):

> And the name of your module say "jsonrpc" but it actually returns "http"
> (obviously using the json cookie management api on controller).
> 
> IMHO It is a good feature but a l think you open some backdoors that can
> not be expose "public".
> 
> If you fix:
> 
> 1.- Not depend of website instead of web.
> 2.- auth="user" and not "public".

**Comment by OSguard** [comment]
We can discuss the name, the idea was to have a documentation what you can have access via jsonrpc and the models are only one endpoint. It depends if it should be list of all JSON data or is just a module to explore models and fields.

I see no diffrents between user and public. Your source is also public available (as it is AGPL) so you don't share any new information. It would be better if you only see objects and fields you have access depends on your current user. But this is not included in the first version.

**Comment by max3903** [comment]
Can you rebase so we can test on runbot ?

**Comment by rafaelbn** [comment]
Interesting for testing when available

**Comment by rafaelbn** [comment]
cc @yajo @pedrobaeza

**Comment by yajo** [comment]
What's the use case for this?

**Comment by OSguard** [comment]
the aim was to have a module to easy explore all models and fields of your current installation.
You have a similar feature in the technical feature but then you can not easy click on relation.

This is morge for develop purpose then for production system.
We try to make it independent from "website" and only use "web" module but then we need to implement much more feature to support a nice template. So we skip this.

**Comment by yajo** [comment]
> but then you can not easy click on relation

What do you mean with this?

**Comment by OSguard** [comment]
Compare:
http://3148243-1-414c31.runbot2.odoo-community.org/doc/jsonrpc/invoice-line-274

with:
http://3148243-1-414c31.runbot2.odoo-community.org/web#id=274&view_type=form&model=ir.model&menu_id=29&action=13

Then you can see you can just click on "account.invoice"

The idea behind was to give a NON-Odoo-Developer a nice view for a XML/JSON-RPC API doku

**Comment by yajo** [comment]
Why not just add this link field in the backend and a portal group with read access to those fields? This seems a little overkill honestly.

**Comment by OSguard** [comment]
This was the first PR for the new website feature almost 2 years ago. It was more a PoC to figure out want we can do with the new Core feature and gives us more opportunities then hacking JS widget for backend.
So i fine if we close this. If someone needs this code. He can find it via odoo-code-search and use it. But we don't need then to maintain this in OCA if no one is interested.

**Comment by yajo** [comment]
I'm not going to vote against this just because I don't need it, but neither will vote up because I think core features are enough for this purpose. Let others talk. Thanks for your understanding :wink:

**Comment by OSguard** [comment]
@rafaelbn @pedrobaeza @dreispt 
can you give a vote if we want this or close the PR?

**Comment by rafaelbn** [comment]
I would like to have the opinion of website fighters. Please @antespi @sergio-teruel @carlosdauden could you vote here?
Any case: if you @OSguard are not using it anymore maybe that is enough yo close it.
Let's see this week whay people say

**Comment by rafaelbn** [comment]
HI @OSguard ,

As nobody answer then my vote is to merge o close depending of your needs. If you are going to maintain and migrate to v9 go on, if not maybe is better to close it.

**Comment by dreispt** [comment]
This is not TraviCI checked, and there are a few guidelines I 'm sure need fixing.
Can you rebase on latest website 8.0?

File: website_doc_jsonrpc/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Website Odoo JSON RPC doc',
+    'version': '0.1',
+    'author': 'initOS GmbH & Co. KG',
+    'category': '',
+    'description': """Generates a documentation for JSON RPC
+    for the current installation on the fly""",
```
Comment (dreispt) [inline website_doc_jsonrpc/__openerp__.py:29]: I'm sorry, it's not clear to me the purpose of this module.

**Comment by rafaelbn** [comment]
Hi, I'm closing this old PR. If needed please make a new PR.

**Pull request closed without merging**